src/tiledb/cloud/udf.py

```diff
@@ -550,14 +551,32 @@ def unshare(name=None, namespace=None, async_req=False):
 """
 
 
-def delete(name, namespace, async_req=False):
+def delete(
+    uri: str, namespace: Optional[str] = None, *, async_req: bool = False
+) -> None:
     """
     Deletes a registered udf
-    :param name: name of udf
-    :param namespace: namespace the udf belongs to
-    :param async_req: return future instead of results for async support
+
+    :param uri: TileDB URI of the udf, defaults to None.
+    :param namespace: namespace the udf belongs to, defaults to None.
+        DEPRECATION WARNING: Will be deprecate from version 0.12.17
+    :param async_req: Return future instead of results for async support
     :return: deleted udf details
     """
+    try:
+        namespace, name = utils.split_uri(uri)
```

Comment:
Does this work correctly for `tiledb://<namespace>/s3://...` URIs?

Reply:
Sort of.
- `namespace` will be correct
- `name` will be the whole `s3://...` path